### PR TITLE
STStatusMonitor refactoring & fixes

### PR DESCRIPTION
**Pull request by virusman**
Replaced duplicated status variables with one source and only pass change notifications to STApplication. Also fixes ping timer/long poller conflict.

### CODE REVIEW

File: syncthing/STStatusMonitor.m [hunk 1]
```diff
@@ -92,6 +93,15 @@ - (void) processEvent:(NSDictionary *)event {
...
+    if (![_syncthing ping]) {
+        self.currentStatus = SyncthingStatusOffline;
+    }
+    else if (self.currentStatus == SyncthingStatusOffline) {
+        self.currentStatus = SyncthingStatusIdle;
+    }
+}
+
 - (void) updateCurrentStatus {
     //NSLog(@"updateCurrentStatus: %@", self.folderStates);
```
Comment (xor-gate) [inline syncthing/STStatusMonitor.m]: probably this log can be removed

**Comment by virusman** [comment]
Removed log call

**Comment by xor-gate** [comment]
If you rebase then the build will work.

**Comment by virusman** [comment]
Done

**Pull request merged**